From bf1fde3dd8dd8be1f8000078fc729977f639f6ef Mon Sep 17 00:00:00 2001
From: Eugen Hristev <eugen.hristev@microchip.com>
Date: Tue, 2 Jun 2020 12:36:13 +0300
Subject: [PATCH 263/471] sdhci: sdhci-of-at91: issue IP reset at probe

Issue IP reset at probe time.
This will avoid clock glitches if input clocks are being requested
again.

Signed-off-by: Eugen Hristev <eugen.hristev@microchip.com>
---
 drivers/mmc/host/sdhci-of-at91.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/sdhci-of-at91.c b/drivers/mmc/host/sdhci-of-at91.c
index 8a68fed024be..f0feaf72c36b 100644
--- a/drivers/mmc/host/sdhci-of-at91.c
+++ b/drivers/mmc/host/sdhci-of-at91.c
@@ -352,6 +352,9 @@ static int sdhci_at91_probe(struct platform_device *pdev)
 	priv = sdhci_pltfm_priv(pltfm_host);
 	priv->soc_data = soc_data;
 
+	/* Perform a software reset before using the IP */
+	sdhci_at91_reset(host, SDHCI_RESET_ALL);
+
 	priv->mainck = devm_clk_get(&pdev->dev, "baseclk");
 	if (IS_ERR(priv->mainck)) {
 		if (soc_data->baseclk_is_generated_internally) {
-- 
2.34.1

